src/conf_mode/interfaces-tinc.py

```diff
@@ -0,0 +1,243 @@
+            render(tinc_host_local_peer_config, 'tinc/hosts_config.tmpl', tinc)
+            call(f'tinc -b -n {network} generate-keys {keys_length}')
+            shutil.move(f'{system_network_dir}/ed25519_key.priv', f'{tinc_network_dir}/ed25519_key.priv')
+            shutil.move(f'{system_network_dir}/rsa_key.priv', f'{tinc_network_dir}/rsa_key.priv')
+        elif not os.path.exists(rsa_private_keyfile) or not os.path.exists(ed25519_private_keyfile) or not os.path.exists(tinc_host_local_peer_config):
+            if os.path.exists(rsa_private_keyfile):
+                os.remove(rsa_private_keyfile)
+            if os.path.exists(ed25519_private_keyfile):
+                os.remove(ed25519_private_keyfile)
+            if os.path.exists(tinc_host_local_peer_config):
+                os.remove(tinc_host_local_peer_config)
+            render(tinc_host_local_peer_config, 'tinc/hosts_config.tmpl', tinc)
+            call(f'tinc -b -n {network} generate-keys {keys_length}')
+            shutil.move(f'{system_network_dir}/ed25519_key.priv', f'{tinc_network_dir}/ed25519_key.priv')
+            shutil.move(f'{system_network_dir}/rsa_key.priv', f'{tinc_network_dir}/rsa_key.priv')
+        else:
+            try:
+                call(f'tinc -n {network} del subnet')
+                call(f'tinc -n {network} del address')
+                call(f'tinc -n {network} del port')
+                if 'subnets' in tinc:
+                    for subnet in tinc['subnets']:
+                        call(f'tinc -n {network} add subnet {subnet}')
+                if 'local_address' in tinc:
+                    for address in tinc['local_address']:
+                        call(f'tinc -n {network} add address {address}')
+                port = tinc['port']
+                call(f'tinc -n {network} add port {port}')
+            except:
+                pass
+        render(tinc_main_config, 'tinc/tinc.conf.tmpl', tinc)
+        if 'connect' in tinc:
+            for conn in tinc['connect']:
+                if 'conf_path' in tinc['connect'][conn]:
+                    conf = f'{tinc_hosts_dir}/{conn}'
+                    system_conf = f'{system_network_dir}/hosts/{conn}'
+                    if tinc['connect'][conn]['conf_path'] != conf and tinc['connect'][conn]['conf_path'] != system_conf:
+                        base = tinc['connect'][conn]['conf_path']
+                        tmp = f'/tmp/remote_tinc_{conn}'
+                        os.symlink(base, conf)
+            
+    return None
+
+def apply(tinc):
+    if tinc is None:
+        return None
+    interface = tinc['ifname']
+    if 'deleted' in tinc or 'disable' in tinc:
+        network = tinc['ifname']
+        call(f'systemctl stop tinc@{network}')
+        if 'deleted' in tinc:
+            create_or_delete_network(tinc, 'delete') # Delete a directory
+        try:
+            t = TincIf(interface)
```

Comment:
I see no implementation of `TincIf()`

Reply:
Seems to be lost in late conflict resolution, I've manually restored this and kept the mainline in sync